source/RemoteWiring/HardwareProfile.h

```diff
@@ -0,0 +1,351 @@
+/*
+    Copyright(c) Microsoft Open Technologies, Inc. All rights reserved.
+
+    The MIT License(MIT)
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files(the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+*/
+
+#pragma once
+
+namespace Microsoft {
+namespace Maker {
+namespace RemoteWiring {
+
+/*
+ * Protocol enum is used to recognize which protocol the initialization data represents.
+ * Currently, the only option is Firmata, which is the only protocol Remote Arduino currently supports
+ */
+public enum class Protocol
+{
+    FIRMATA
+};
+
+/*
+ * Pin capabilities are stored as bitmasks, the PinCapability enum represents the bit value of each capability.
+ */
+public enum class PinCapability
```

Comment:
These are Firmata specific capabilities. I was going to suggest having that reflected in the name, but the implementation seems tied to it. This makes me feel like we need a hardware profile interface with Firmata as an implementation.